detection_rules/utils.py

```diff
@@ -528,3 +528,159 @@ def get_identifiers(self) -> list[str]:
                 # another group we're not expecting
                 raise ValueError("Unrecognized named group in pattern", self.pattern)
         return ids
+
+
+FROM_SOURCES_REGEX = re.compile(r"^\s*FROM\s+(?P<sources>.+?)\s*(?:\||\bmetadata\b|//|$)", re.IGNORECASE | re.MULTILINE)
+
+
+def get_esql_query_indices(query: str) -> tuple[str, list[str]]:
+    """Extract indices from an ES|QL query."""
+    match = FROM_SOURCES_REGEX.search(query)
+
+    if not match:
+        return "", []
+
+    sources_str = match.group("sources")
+    return sources_str, [source.strip() for source in sources_str.split(",")]
+
+
+# NOTE This is done with a dataclass but could also be done with dict, etc.
+# Also other places this is called an <integration.integration> instead.
+# such as in integrations.py def parse_datasets
```

Comment:
Do we want to be consistent then? Is this package/integrations?

Reply:
Peer review, switching to package/integrations as this is more consistent with the UI.